Component/ODESolver/Backward/tests/EulerImplicitSolverStatic_test.cpp

```diff
@@ -25,33 +25,69 @@ using sofa::testing::BaseSimulationTest;
 #include <sofa/testing/NumericTest.h>
 using sofa::testing::NumericTest;
 
-#include <SceneCreator/SceneCreator.h>
-#include <SceneCreator/SceneUtils.h>
-#include <SofaImplicitOdeSolver/EulerImplicitSolver.h>
-#include <SofaBaseLinearSolver/CGLinearSolver.h>
-#include <SofaBaseMechanics/UniformMass.h>
-#include <SofaBaseMechanics/MechanicalObject.h>
-#include <SofaBoundaryCondition/FixedConstraint.h>
-#include <SofaDeformable/StiffSpringForceField.h>
+#include <SofaSimulationGraph/SimpleApi.h>
 
 #include <sofa/simulation/Simulation.h>
+#include <sofa/component/odesolver/testing/EigenTestUtilities.h>
 
 
 namespace sofa {
 
-using namespace modeling;
 using namespace type;
 using namespace testing;
 using namespace defaulttype;
 using core::objectmodel::New;
 
-using sofa::component::mass::UniformMass;
-using sofa::component::container::MechanicalObject;
-using sofa::component::interactionforcefield::StiffSpringForceField;
-using sofa::component::projectiveconstraintset::FixedConstraint;
-using sofa::component::odesolver::EulerImplicitSolver;
-typedef component::linearsolver::CGLinearSolver<component::linearsolver::GraphScatteredMatrix, component::linearsolver::GraphScatteredVector> CGLinearSolver;
 
+/// Create a stiff string
+Node::SPtr massSpringString(Node::SPtr parent,
+    double x0, double y0, double z0, // start point,
+    double x1, double y1, double z1, // end point
+    unsigned numParticles,
+    double totalMass,
+    double stiffnessValue,
+    double dampingRatio)
+{
+    static unsigned numObject = 1;
+    std::ostringstream oss;
+    oss << "string_" << numObject++;
+
+    Vec3d startPoint(x0, y0, z0), endPoint(x1, y1, z1);
+    SReal totalLength = (endPoint - startPoint).norm();
+
+    std::stringstream positions;
+    std::stringstream springs;
+    for (unsigned i = 0; i < numParticles; i++)
+    {
+        double alpha = (double)i / (numParticles - 1);
+        Vec3d currpos = startPoint * (1 - alpha) + endPoint * alpha;
+        positions << simpleapi::str(currpos) << " ";
+
+        if (i > 0)
+        {
+            springs << simpleapi::str(i - 1) << " " << simpleapi::str(i) << " " << simpleapi::str(stiffnessValue)
+                << " " << simpleapi::str(dampingRatio) << " " << simpleapi::str(totalLength / (numParticles - 1));
+        }
+    }
+
+    Node::SPtr node = simpleapi::createChild(parent, oss.str());
+    simpleapi::createObject(node, "MechanicalObject", {
+                                {"name", oss.str() + "_DOF"},
+                                {"size", simpleapi::str(numParticles)},
+                                {"position", positions.str()}
+        });
+
+    simpleapi::createObject(node, "UniformMass", {
+                                {"name",oss.str() + "_mass"},
+                                {"vertexMass", simpleapi::str(totalMass / numParticles)} });
+
+    simpleapi::createObject(node, "StiffSpringForceField", {
```

Comment:
https://github.com/sofa-framework/sofa/issues/417
In a nutshell (and IMO) it will deport the dependencies,etc at runtime, so the test project itself does not need a lot of deps. 

In my mind this is like "writing a XML file" so if your composite-object is fine with XML, it should be fine with simpleapi (if it works as I understood...)

Reply:
Ok, I'll check that in https://github.com/sofa-framework/sofa/pull/2554 when possible